pkg/route/graph/analysis/analysis.go

```diff
@@ -70,6 +75,54 @@ route:
 
 				continue route
 			}
+
+			if routeNode.Spec.Port != nil {
+				routePortString := routeNode.Spec.Port.TargetPort.String()
+				if routePort, err := strconv.Atoi(routePortString); err == nil {
+					// route has numeric port, service has no port with that number as a targetPort.
```

Comment:
second part of this comment is misleading and only valid after the following loop